### PR TITLE
[NetteUtilsCodeQuality] Fixes #4686 Add SubstrMinusToStringEndsWithRector

### LINKED ISSUE #4686: [NetteUtilsCodeQuality] Add substr -minus value to Strings - PHPStan or Rector rule?
```diff
-if (substr($shortClassName, -4) !== 'Test') {
+if (! Strings::endsWith($shortClassName, 'Test')) {
```

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
@TomasVotruba it is implemented, composer error seems unrelated: 

```bash
composer install --no-progress --ansi

  Problem 1
    - Root composer.json requires vendor/vendor-patches, it could not be found in any version, there may be a typo in the package name.
```

File: rules/nette-code-quality/tests/Rector/Identical/SubstrMinusToStringEndsWithRector/Fixture/with_var_value.php.inc [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Rector\NetteCodeQuality\Tests\Rector\Identical\SubstrMinusToStringEndsWithRector\Fixture;
+
+function withVarValue($var, $value)
+{
+    substr($var, -4) !== $value;
+    $value !== substr($var, -4);
+    substr($var, -4) === $value;
+    $value === substr($var, -4);
```
Comment (TomasVotruba) [inline rules/nette-code-quality/tests/Rector/Identical/SubstrMinusToStringEndsWithRector/Fixture/with_var_value.php.inc:10]: There should be a skipped test fixture with different number than word length.




```php
'part' !== substr($var, -3);

**Comment by TomasVotruba** [comment]
> vendor/vendor-patches

That's a typo I probably named during rename.
Fix to `symplify/vendor-patches` should help

**Comment by TomasVotruba** [comment]
I fixed it on `master` https://github.com/rectorphp/rector/commit/4c9b816537fff2fd6f019f2d8e964d5f87d11821

Works well :+1:

File: rules/nette-code-quality/tests/Rector/Identical/SubstrMinusToStringEndsWithRector/Fixture/with_var_value.php.inc (hunk 1 above)
Comment (samsonasik) [reply rules/nette-code-quality/tests/Rector/Identical/SubstrMinusToStringEndsWithRector/Fixture/with_var_value.php.inc:10]: implemented if comparison with string, check length of string first.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by TomasVotruba** [comment]
Looks good, thank you :+1:

**Pull request merged**